Pod/Classes/WPPHAssetDataSource.m

```diff
@@ -119,61 +119,70 @@ - (void)photoLibraryDidChange:(PHChange *)changeInstance
 {
-    PHAuthorizationStatus status = [PHPhotoLibrary authorizationStatus];
-    switch (status) {
-        case PHAuthorizationStatusRestricted:
-        {
-            if (failureBlock) {
-                NSError *error = [NSError errorWithDomain:WPMediaPickerErrorDomain code:WPMediaPickerErrorCodeRestricted userInfo:nil];
-                failureBlock(error);
-            }
-            return;
-        }
-        case PHAuthorizationStatusDenied:
-        {
-            if (failureBlock) {
-                NSError *error = [NSError errorWithDomain:WPMediaPickerErrorDomain code:WPMediaPickerErrorCodePermissionDenied userInfo:nil];
-                failureBlock(error);
+    [self checkPermissionStatus:^(PHAuthorizationStatus status) {
+        switch (status) {
+            case PHAuthorizationStatusRestricted:
+            {
+                if (failureBlock) {
+                    NSError *error = [NSError errorWithDomain:WPMediaPickerErrorDomain code:WPMediaPickerErrorCodeRestricted userInfo:nil];
+                    failureBlock(error);
+                }
+                return;
             }
-            return;
-        }
-        case PHAuthorizationStatusNotDetermined:
-        {
-            [PHPhotoLibrary requestAuthorization:^(PHAuthorizationStatus status) {
-                [self loadDataWithOptions:options success:successBlock failure:failureBlock];
-            }];
-            return;
-        }
-        case PHAuthorizationStatusAuthorized: {
-            dispatch_async(dispatch_get_global_queue(QOS_CLASS_USER_INITIATED, 0), ^{
-                if (self.activeAssetsCollection == nil) {
-                    self.activeAssetsCollection = [[PHAssetCollection fetchAssetCollectionsWithType:PHAssetCollectionTypeSmartAlbum
-                                                                                            subtype:PHAssetCollectionSubtypeSmartAlbumUserLibrary
-                                                                                            options:nil] firstObject];
+            case PHAuthorizationStatusDenied:
+            {
+                if (failureBlock) {
+                    NSError *error = [NSError errorWithDomain:WPMediaPickerErrorDomain code:WPMediaPickerErrorCodePermissionDenied userInfo:nil];
+                    failureBlock(error);
                 }
-                switch (options) {
-                    case (WPMediaLoadOptionsGroups): {
-                        [self loadGroupsWithSuccess:successBlock failure:failureBlock];
-                        return;
-                    }
-                    case (WPMediaLoadOptionsAssets): {
-                        [self loadAssetsWithSuccess:successBlock failure:failureBlock];
-                        return;
+                return;
+            }
+            case PHAuthorizationStatusNotDetermined:
+            {
+                [PHPhotoLibrary requestAuthorization:^(PHAuthorizationStatus status) {
```

Comment:
I may not be following the code correctly, but should we be doing the iOS 14 check here again, or not? How come we don't call `requestAuthorizationForAccessLevel` again here? 🤔

Reply:
You are absolutely correct! Just updated the code.